### DISCUSSION TITLE
Should have an additional example in webhook or subscription to unsubscribe previous

### DISCUSSION THREAD

**Discussion by klausr3** [pay-rails/pay#481] [General] [answered]
I think could be a good idea an example of a webhook/subscription logic to unsubscribe previous subscription.
When do you have a list of plan/product and each one with monthly/yearly prices, the client may want to 'upgrade' from Basic plan to Premium plan, I think it should be handled in a webhook, when the Premium has successfull paid, the Basic subscription should be automatically canceled by a webhook logic, avoiding user to keep paying for Basic plan and Premium plan. I think it should be nice to have this example in docs. I'm working on it right now. (using stripe)

**Comment by excid3** [comment, marked as answer]
Hmm, that's not a standard way of doing things. You would just change the subscription plan/price from Basic to Premium instead of creating a whole new subscription.

Edit: Use the `@pay_subscription.swap("new_price_id")` method. https://github.com/pay-rails/pay/blob/master/docs/6_subscriptions.md#swap-a-subscription-to-another-plan

**Reply by klausr3** [reply]
Thanks @excid3
I didn't know there was this option to change the subscription price.

**Reply by excid3** [reply]
Yep! Check out the `swap` method on the subscription. It is used for changing the subscriptions plan/price. 👍 

Here's the section for [`swap` in the Pay docs](https://github.com/pay-rails/pay/blob/master/docs/6_subscriptions.md#swap-a-subscription-to-another-plan).
